contrib/utilities/jsontomarkdown.py

```diff
@@ -97,6 +97,9 @@ def escape_doc_string(text) :
     tmp = re.sub(r'\\texttt\{(.*?)\}',
                  r'`\1`',
                  tmp)
+    tmp = re.sub(r'\\note\{(.*?)\}',
```

Comment:
I think this only works if this replacement happens after the three (four with \cite from #5450) happen first because they can appear inside a note. Can you add a comment before this line stating that this replacement should stay at the bottom?

Reply:
This is all very fragile because it doesn't allow for any occurrence of curly braces within the note. This is of course pre-existing for the other parts. There is really no good solution for this. I wished there was a way to write regexes so that they handle nested braces (or any kind of parentheses)...

Reply:
If we ever wanted to do better: https://stackoverflow.com/questions/14952113/how-can-i-match-nested-brackets-using-regex
http://www.drregex.com/2017/11/match-nested-brackets-with-regex-new.html
There's some real gems in terms of regexes in these posts :-)

Reply:
Maybe we should restrict regex replacements to simple cases (italics, bold, etc.) where this is unlikely to happen and manually replace all the ``\note`` by the sphinx command.

Regardless, I think the current solution is good enough and an improvement.

Reply:
Yes, I agree. All I wanted to point out is that there is a problem. I didn't mean to suggest that it needs to be addressed right now.